pkg/test/framework/components/cluster/common.go

```diff
@@ -0,0 +1,98 @@
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package cluster
+
+import (
+	"fmt"
+
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+func NewTopology(
+	name string,
+	networkName string,
+	controlPlaneCluster string,
+	configCluster string,
+	clusters map[string]resource.Cluster,
+) Topology {
+	return Topology{
+		name:                name,
+		networkName:         networkName,
+		controlPlaneCluster: controlPlaneCluster,
+		configCluster:       configCluster,
+		clusters:            clusters,
+	}
+}
+
+// Topology gives information about the relationship between clusters.
+// Cluster implementations can embed this struct to include common functionality.
+type Topology struct {
+	name                string
+	networkName         string
+	controlPlaneCluster string
+	configCluster       string
+	// clusters should contain all clusters in the context
+	clusters map[string]resource.Cluster
+}
+
+// NetworkName the cluster is on
+func (c Topology) NetworkName() string {
+	return c.networkName
+}
+
+// Name provides the name this cluster used by Istio.
+func (c Topology) Name() string {
```

Comment:
The idea is that we'd embed this in most `Cluster` implementations to avoid a ton of copy-pasted logic. We already have `cluster.Name`, so I didn't think we needed to change it. The fact that this is in the `cluster` package + comments should indicate this is a component of cluster.

Reply:
Sure, that's fine.